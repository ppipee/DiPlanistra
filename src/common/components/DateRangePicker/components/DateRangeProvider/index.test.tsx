import React from 'react'

import 'jest-styled-components'
import { render } from '@testing-library/react'

import * as DateRangeProviderType from '.'

describe('Test DateRangeProvider', () => {
	const DateRangeContextSpy = jest.fn()
	const useToggleSpy = jest.fn()
	const selectControllerSpy = jest.fn()

	jest.doMock('common/components/DateRangePicker/context', () => ({
		Provider: (props: Record<string, any>) => {
			DateRangeContextSpy(props)

			return <div {...props}>{props.children}</div>
		},
	}))

	const { default: DateRangeProvider } = require('.') as typeof DateRangeProviderType

	afterEach(() => {
		jest.resetAllMocks()
	})

	it('should render DateRangeProvider with style correctly', () => {
		useToggleSpy.mockReturnValue([false, selectControllerSpy])

		const { container } = render(
			<DateRangeProvider date={'Mock Date' as any} setDate={jest.fn() as any}>
				<div />
			</DateRangeProvider>,
		)

		expect(container.firstChild).toMatchSnapshot()
	})
})
