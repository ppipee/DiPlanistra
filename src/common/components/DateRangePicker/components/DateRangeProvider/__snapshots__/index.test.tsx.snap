// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DateRangeProvider should render DateRangeProvider with style correctly 1`] = `
<div
  value="[object Object]"
>
  <div />
</div>
`;
